cmake/configure/configure_3_petsc.cmake

```diff
@@ -103,6 +103,20 @@ MACRO(FEATURE_PETSC_CONFIGURE_EXTERNAL)
   SET(DEAL_II_EXPAND_PETSC_BLOCKVECTOR "PETScWrappers::BlockVector")
   SET(DEAL_II_EXPAND_PETSC_MPI_VECTOR "PETScWrappers::MPI::Vector")
   SET(DEAL_II_EXPAND_PETSC_MPI_BLOCKVECTOR "PETScWrappers::MPI::BlockVector")
+  #
+  # FIXME:
+  # temporary variable until deal.II fully support complex-valued PETSc
+  # Note that petscconf.h should export PETSC_USE_COMPLEX 1 in complex-valued case.
+  FILE(READ ${PETSC_INCLUDE_DIRS}/petscconf.h _petscconf)
+  STRING(REGEX MATCH "\#define PETSC_USE_COMPLEX 1" _petsc_use_complex ${_petscconf})
+  IF( NOT _petsc_use_complex )
+    SET(DEAL_II_EXPAND_PETSC_VECTOR_REAL "PETScWrappers::Vector")
+    SET(DEAL_II_EXPAND_PETSC_BLOCKVECTOR_REAL "PETScWrappers::BlockVector")
+    SET(DEAL_II_EXPAND_PETSC_MPI_VECTOR_REAL "PETScWrappers::MPI::Vector")
+    SET(DEAL_II_EXPAND_PETSC_MPI_BLOCKVECTOR_REAL "PETScWrappers::MPI::BlockVector")
+  ELSE()
+    MESSAGE(STATUS "Compiling with complex-valued algebra")
+  ENDIF()
 ENDMACRO()
```

Comment:
The configure file ```modules/FindPETSC.cmake``` should already export ```PETSC_WITH_COMPLEX```:

```
 81   FILE(STRINGS "${PETSC_PETSCCONF_H}" PETSC_COMPLEX_STRING                       
 82     REGEX "#define.*PETSC_USE_COMPLEX 1")                                        
 83   IF("${PETSC_COMPLEX_STRING}" STREQUAL "")                                      
 84     SET(PETSC_WITH_COMPLEX FALSE)                                                
 85   ELSE()                                                                         
 86     SET(PETSC_WITH_COMPLEX TRUE)                                                 
 87   ENDIF() 
```

Can you refactor to use this instead of ```_petsc_use_complex```? :-)